### PR TITLE
ENG-3243: redis cli and utils

### PR DESCRIPTION
**Pull request by kevinszuchet**

### CODE REVIEW

File: src/cache/redis/locking.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { makeLockUnlockWithId, withLockByInput } = require("gamlajs").default;
+const { pipe, curryN } = require("ramda");
+
+const distributedLock = curryN(4, (argsToLockId, lockIdToKey, redisClient, f) =>
```
Comment (nitzo) [inline src/cache/redis/locking.js:4]: did we have tests for this ?
if not - can we open a task to create tests ?

**Review by nitzo — APPROVED** [review]

File: src/cache/redis/RedisClient.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const redis = require("redis");
+const { promisifyRedisClient } = require("./utils");
+
+const makeNewRedisClient = (redisConfig) => redis.createClient(redisConfig);
```
Comment (davidstrouk) [inline src/cache/redis/RedisClient.js:4]: maybe rename to `createRedisClient`, to be consistent with the inner function call `redis.createClient` ? not critical though, uty

File: src/cache/redis/locking.js (hunk 1 above)
Comment (davidstrouk) [inline src/cache/redis/locking.js:4]: as discussed please talk with @romCohen and understand why do we have a need for two different funcs here `argsToLockId` and `lockIdToKey`

**Review by davidstrouk — APPROVED** [review]

File: src/cache/redis/locking.js (hunk 1 above)
Comment (davidstrouk) [reply src/cache/redis/locking.js]: @kevinszuchet do we have an answer re this?

Comment (kevinszuchet) [reply src/cache/redis/locking.js]: "Basically you have 2 functions here, one to generate the lock id in redis. the second to get the redis key from the lockId"

**Pull request merged**